modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/Walker.java

```diff
@@ -251,7 +252,20 @@ public ANode visitSource(SourceContext ctx) {
             statements.add((AStatement)visit(statement));
         }
 
-        return new SClass(scriptClassInfo, sourceName, sourceText, debugStream, location(ctx), functions, statements);
+        String returnCanonicalTypeName = PainlessLookupUtility.typeToCanonicalTypeName(scriptClassInfo.getExecuteMethodReturnType());
+        List<String> paramTypes = new ArrayList<>();
+        List<String> paramNames = new ArrayList<>();
+
+        for (ScriptClassInfo.MethodArgument argument : scriptClassInfo.getExecuteArguments()) {
+            paramTypes.add(PainlessLookupUtility.typeToCanonicalTypeName(argument.getClazz()));
+            paramNames.add(argument.getName());
+        }
+
+        SFunction execute = new SFunction(location(ctx), returnCanonicalTypeName, "execute", paramTypes, paramNames, new SBlock(
+                location(ctx), statements), true, false, false, true);
```

Comment:
>  true, false, false, true

ugh.

Reply:
Agreed. I would like to change these to an enum in a future PR.